From d16a4036137654e2dfe3ec8ee2221f51174b59f4 Mon Sep 17 00:00:00 2001
From: Lorenzo Pirritano <lpirritano@anaconda.com>
Date: Fri, 7 Jun 2024 15:10:18 +0200
Subject: [PATCH 2/2] unvendor_boost_and_eigen

Also remove:
  - lib/boost_1_84_0.tar.gz
  - lib/boost
  - lib/eigen

---
 setup.py | 34 ++++++++++++++++++++++++----------
 1 file changed, 24 insertions(+), 10 deletions(-)

diff --git a/setup.py b/setup.py
index ce9de18..7e8aa45 100644
--- a/setup.py
+++ b/setup.py
@@ -1,17 +1,31 @@
 import os
-import tarfile
-from glob import glob
+import shutil
 from pathlib import Path
 
 from pybind11.setup_helpers import Pybind11Extension, build_ext
 from setuptools import setup
 
-# Extract boost if not already done.
-if not os.path.isdir('lib/boost'):
-  boost_archive = glob(os.path.join('lib', 'boost*.tar.gz'))[0]
-  tar = tarfile.open(boost_archive)
-  tar.extractall(path='lib/boost')
-  tar.close()
+# remove lib/boost_1_84_0.tar.gz
+local_lib_boost_tarball = os.path.join("lib", "boost_1_84_0.tar.gz")
+if not os.path.exists(local_lib_boost_tarball):
+    os.remove(local_lib_boost_tarball)
+# remove folder lib/boost which is not empty
+local_lib_boost_dir = os.path.join("lib", "boost")
+if not os.path.isdir(local_lib_boost_dir):
+    shutil.rmtree(local_lib_boost_dir)
+# remove folder lib/eigen which is not empty
+local_lib_eigen_dir = os.path.join("lib", "eigen")
+if not os.path.isdir(local_lib_eigen_dir):
+    shutil.rmtree(local_lib_eigen_dir)
+
+include_dir = os.environ.get("LIBRARY_INC", None)
+# On Mac and Linux LIBRARY_INC is not defined
+if include_dir is None:
+    include_dir = os.path.join(os.environ.get("PREFIX"), 'include')
+
+boost_include_dir = os.path.join(include_dir, 'boost')
+eigen_include_dir = os.path.join(include_dir, 'eigen3')
+eigen_unsupported_include_dir = os.path.join(include_dir, 'eigen', 'unsupported')
 
 setup(
     name='pyvinecopulib',
@@ -20,8 +34,8 @@ setup(
     ext_modules=[
         Pybind11Extension('pyvinecopulib', ['src/main.cpp'],
                           include_dirs=[
-                              'lib/boost', 'lib/eigen',
-                              'lib/eigen/unsupported',
+                              boost_include_dir, eigen_include_dir,
+                              eigen_unsupported_include_dir,
                               'lib/vinecopulib/include', 'lib/wdm/include'
                           ],
                           language='c++',
-- 
2.39.1

